src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -691,7 +710,7 @@ make_DirectStepI (
 {
     // Only charge a transfer fee if the previous step redeems
     auto r = std::make_unique<DirectStepI> (
-        src, dst, c, ctx.prevStep, ctx.j);
+        src, dst, c, ctx.prevStep, ctx.isLast, ctx.j);
```

Comment:
The correct state needs be pulled out of a `StrandContext` only once. We can do it in the constructor, or we can do it in `make_DirectStepI`. Moving it to the constructor is fine with me, but not for this PR.